src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/PackageVulnerabilityService.cs

```diff
@@ -17,26 +18,26 @@ namespace NuGet.PackageManagement.VisualStudio
     public class PackageVulnerabilityService : IPackageVulnerabilityService
     {
         private readonly IEnumerable<SourceRepository> _sourceRepositories;
-        private GetVulnerabilityInfoResult _vulnerabilities;
+        private readonly AsyncLazy<GetVulnerabilityInfoResult> _vulnerabilitiesLazy;
         private INuGetUILogger _logger;
 
-        public PackageVulnerabilityService(IEnumerable<SourceRepository> sourceRepositories, INuGetUILogger logger)
+        public PackageVulnerabilityService(IEnumerable<SourceRepository> sourceRepositories, INuGetUILogger logger, CancellationToken cancellationToken)
         {
             _sourceRepositories = sourceRepositories ?? throw new ArgumentNullException(nameof(sourceRepositories));
-            _logger = logger ?? throw new ArgumentNullException(nameof(logger)); ;
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _vulnerabilitiesLazy = new AsyncLazy<GetVulnerabilityInfoResult>(async () => await GetAllVulnerabilityDataAsync(CancellationToken.None));
```

Comment:
It's good that `GetAllVulnerabilityDataAsync` isn't cancelled when a customer changes the search filter, or changes tabs in PM UI.  However, if PM UI is closed, but this call is still pending, it should be cancelled.

I think an easy way to do this with the current design is to make this service class disposable, make sure that PM UI disposes it on close, and then have a CancellationTokenSource that is set on dispose.